src/MemoryTracing/MemoryTracingUtils.cpp

```diff
@@ -55,75 +49,106 @@ SystemMemoryUsage ParseMemInfo(std::string_view meminfo_content) {
+      system_memory_usage->set_pgmajfault(value);
+    }
+  }
 }
 
 ErrorMessageOr<SystemMemoryUsage> GetSystemMemoryUsage() noexcept {
-  uint64_t current_timestamp_ns = orbit_base::CaptureTimestampNs();
-
+  SystemMemoryUsage system_memory_usage;
+  system_memory_usage.set_timestamp_ns(orbit_base::CaptureTimestampNs());
+  system_memory_usage.set_total_kb(kMissingInfo);
+  system_memory_usage.set_free_kb(kMissingInfo);
+  system_memory_usage.set_available_kb(kMissingInfo);
+  system_memory_usage.set_buffers_kb(kMissingInfo);
+  system_memory_usage.set_cached_kb(kMissingInfo);
+  system_memory_usage.set_pgfault(kMissingInfo);
+  system_memory_usage.set_pgmajfault(kMissingInfo);
+
+  // Extract system-wide memory usage from the /proc/meminfo file.
   ErrorMessageOr<std::string> reading_result = orbit_base::ReadFileToString("/proc/meminfo");
   if (reading_result.has_error()) {
     ERROR("%s", reading_result.error().message());
     return reading_result.error();
   }
+  GetValuesFromMemInfo(reading_result.value(), &system_memory_usage);
 
-  SystemMemoryUsage parsing_result = ParseMemInfo(reading_result.value());
-  parsing_result.set_timestamp_ns(current_timestamp_ns);
-  return parsing_result;
-}
-
-int64_t GetVmRssFromProcessStatus(std::string_view status_content) {
-  if (status_content.empty()) return kMissingInfo;
-
-  std::vector<std::string> lines = absl::StrSplit(status_content, '\n');
-  if (lines.empty()) return kMissingInfo;
-
-  for (std::string_view line : lines) {
-    if (absl::StartsWith(line, "VmRSS")) {
-      std::vector<std::string> splits = absl::StrSplit(line, ' ', absl::SkipWhitespace{});
-      // According to the kernel code
-      // https://git.kernel.org/pub/scm/linux/kernel/git/torvalds/linux.git/tree/fs/proc/task_mmu.c,
-      // the size unit of the VmRSS field in the /proc/<PID>/status file is fixed to "kB", which
-      // implies 1024 Bytes. We report in "KB" in ProcessSystemUsage.
-      int64_t rss_kb;
-      if (splits.size() == 3 && splits[2] == "kB" && absl::SimpleAtoi(splits[1], &rss_kb)) {
-        return rss_kb;
-      }
-      return kMissingInfo;
-    }
+  // Extract system-wide page fault statistics from the /proc/vmstat file.
+  reading_result = orbit_base::ReadFileToString("/proc/vmstat");
+  if (reading_result.has_error()) {
+    ERROR("%s", reading_result.error().message());
+    return reading_result.error();
   }
+  GetValuesFromVmStat(reading_result.value(), &system_memory_usage);
 
-  return kMissingInfo;
+  return system_memory_usage;
+}
+
+void GetValuesFromProcessStat(std::string_view stat_content,
+                              ProcessMemoryUsage* process_memory_usage) {
+  if (stat_content.empty()) return;
+
+  // According to the kernel code
+  // https://git.kernel.org/pub/scm/linux/kernel/git/torvalds/linux.git/tree/fs/proc/array.c,
+  // the /proc/<PID>/stat file records 52 process status information on a single line, in a fixed
+  // order. We are intersted in the following fields:
```

Comment:
nit: "interested"

Reply:
Done! Thanks for pointing out this typo.